translator/src/translator.rs

```diff
@@ -0,0 +1,450 @@
+                writeln!(self.output, ";");
+            },
+            Statement::Raise(r) => self.visit_raise(r),
+            Statement::BreakStmt(b) => self.visit_break(b),
+            Statement::ContinueStmt(c) => self.visit_continue(c),
+            Statement::Global(g) => self.visit_global(g),
+            Statement::Nonlocal(n) => self.visit_nonlocal(n),
+            Statement::IfStatement(i) => self.visit_if(i),
+            Statement::WhileStatement(w) => self.visit_while(w),
+            Statement::ForStatement(f) => self.visit_for(f),
+            Statement::WithStatement(w) => self.visit_with(w),
+            Statement::TryStatement(t) => self.visit_try(t),
+            Statement::TryStarStatement(t) => self.visit_try_star(t),
+            Statement::FunctionDef(f) => self.visit_function_def(f),
+            Statement::ClassDef(c) => self.visit_class_def(c),
+            Statement::MatchStmt(m) => self.visit_match(m),
+            Statement::AsyncForStatement(f) => self.visit_async_for(f),
+            Statement::AsyncWithStatement(w) => self.visit_async_with(w),
+            Statement::AsyncFunctionDef(f) => self.visit_async_function_def(f),
+            Statement::TypeAlias(a) => self.visit_type_alias(a),
+        }
+    }
+
+    fn visit_expr(&mut self, e: &Expression) {
+        match e {
+            Expression::Constant(c) => self.visit_constant(c),
+            Expression::List(l) => self.visit_list(l),
+            Expression::Tuple(t) => self.visit_tuple(t),
+            Expression::Dict(d) => self.visit_dict(d),
+            Expression::Set(s) => self.visit_set(s),
+            Expression::Name(n) => self.visit_name(n),
+            Expression::BoolOp(b) => self.visit_bool_op(b),
+            Expression::UnaryOp(u) => self.visit_unary_op(u),
+            Expression::BinOp(b) => self.visit_bin_op(b),
+            Expression::NamedExpr(n) => self.visit_named_expr(n),
+            Expression::Yield(y) => self.visit_yield(y),
+            Expression::YieldFrom(y) => self.visit_yield_from(y),
+            Expression::Starred(s) => self.visit_starred(s),
+            Expression::Generator(g) => self.visit_generator(g),
+            Expression::ListComp(l) => self.visit_list_comp(l),
+            Expression::SetComp(s) => self.visit_set_comp(s),
+            Expression::DictComp(d) => self.visit_dict_comp(d),
+            Expression::Attribute(a) => self.visit_attribute(a),
+            Expression::Subscript(s) => self.visit_subscript(s),
+            Expression::Slice(s) => self.visit_slice(s),
+            Expression::Call(c) => self.visit_call(c),
+            Expression::Await(a) => self.visit_await(a),
+            Expression::Compare(c) => self.visit_compare(c),
+            Expression::Lambda(l) => self.visit_lambda(l),
+            Expression::IfExp(i) => self.visit_if_exp(i),
+            Expression::JoinedStr(j) => self.visit_joined_str(j),
+            Expression::FormattedValue(f) => self.visit_formatted_value(f),
```

Comment:
why are the input variable name to each expression all diff 🤣 can they all be `expr`?

Reply:
I copied the names from the interface https://github.com/pcmoritz/enderpy/blob/main/typechecker/src/ast_visitor.rs, but I think it makes sense, since they all have different types (and in particular are not of type Expression -- which is an enum type that contains all of them) so it is better to also use different letters.